src/land/imsfv3_scf2ioda.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python3
+#
+# (C) Copyright 2021 NOAA/NWS/NCEP/EMC
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+
+import sys
+import argparse
+import netCDF4 as nc
+import numpy as np
+import re
+from datetime import datetime, timedelta
+import os
+from pathlib import Path
+
+IODA_CONV_PATH = Path(__file__).parent/"@SCRIPT_LIB_PATH@"
+if not IODA_CONV_PATH.is_dir():
+    IODA_CONV_PATH = Path(__file__).parent/'..'/'lib-python'
+sys.path.append(str(IODA_CONV_PATH.resolve()))
+
+import ioda_conv_engines as iconv
+from collections import defaultdict, OrderedDict
+from orddicts import DefaultOrderedDict
+
+locationKeyList = [
+    ("latitude", "float"),
+    ("longitude", "float"),
+    ("altitude", "float"),
+    ("datetime", "string")
+]
+
+obsvars = {
+    'snow_cover_fraction_': 'snowCoverFraction',
+    'total_snow_depth': 'totalSnowDepth',
+}
+
+AttrData = {
+    'converter': os.path.basename(__file__),
+}
+
+DimDict = {
+}
+
+VarDims = {
+    'snowCoverFraction': ['nlocs'],
+    'totalSnowDepth': ['nlocs'],
+}
+
+
+class imsFV3(object):
+
+    def __init__(self, filename):
+        self.filename = filename
+        self.varDict = defaultdict(lambda: defaultdict(dict))
+        self.outdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+        self.varAttrs = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+        self._read()
+
+    def _read(self):
+
+        # set up variable names for IODA
+        for iodavar in ['snowCoverFraction', 'totalSnowDepth']:
+            self.varDict[iodavar]['valKey'] = iodavar, iconv.OvalName()
+            self.varDict[iodavar]['errKey'] = iodavar, iconv.OerrName()
+            self.varDict[iodavar]['qcKey'] = iodavar, iconv.OqcName()
+            self.varAttrs[iodavar, iconv.OvalName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OerrName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OqcName()]['coordinates'] = 'longitude latitude'
+            if iodavar == 'snowCoverFraction':
+                self.varAttrs[iodavar, iconv.OvalName()]['units'] = 'unitless'
+                self.varAttrs[iodavar, iconv.OerrName()]['units'] = 'unitless'
```

Comment:
This is the message I got from Steve: I agree with Cory on explicitly doing the "unitless" assignment for now. Telling people that all variables need a units attribute, while providing a somewhat hidden list of defaults is confusing and until we come up with a cleaner solution I think it's more clear for people to explicitly set attributes.  I agree his message. I can modify it in the future when the document is ready and we decide to change all similar variables. Thank you.

Reply:
All, can I keep snow cover fraction unit as 'unitless' right now but may be change it in the future or do I need to put it as '1'? Please let me know so that I can work for that part. I am not sure if we have reached a agreement for that part. Just wait for your decision.

Reply:
For unit-less variables, the units depend on the variable.

1. If the variable is categorical, then the units are "unitless". A categorical variable is one that contains numbers that represent types such as land types (forest, farm, etc) or qc flags.

2. If the variable is not categorical, then if it has "hidden" units such as mixing ratio with "kg kg-1", then set the units to those hidden units (ie, mixing ratio uses units set to "kg kg-1").

3. If the variable is not categorical, and it is simply a numeric value such as relative humidity expressed as a fraction between 0 and 1, then set the units to '1'.

I'm guessing that snow cover fraction falls into the classification of "non-categorical and simply a numeric value" (number 3 in the above list). Ie, there are no hidden units such as "kg kg-1". Is this correct? If so, then please change the units to '1'. Thanks!

Reply:
Thank you, @srherbener. I changed it as 1 as requested. Thank you.

Reply:
@amfox37 and @ClaraDraper-NOAA , note that snow cover fraction unit will be  "1" rather than 'unitless'.